Reddit/Reddit/Controllers/HomeController.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+using Reddit.Models;
+using Reddit.Servises;
+using Reddit.ViewModel;
+using System;
+using System.Diagnostics;
+
+namespace Reddit.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly ILogger<HomeController> logger;
+        private readonly IPostService postService;
+
+        public HomeController(ILogger<HomeController> logger, IPostService postService)
+        {
+            this.logger = logger;
+            this.postService = postService;
+        }
+
+
+        public IActionResult Index()
+        {
+            var posts = postService.Gell10BestPost();
+            return View(new PostViewModel
+            {
+                Posts = posts
+            });
+        }
+
+
+        public IActionResult List(string searchString)
+        {
+
+            var posts = postService.GetAllPost();
```

Comment:
I would make this after null check, because you will load them and if the string is not null, you dont use them. But the cost of loading data from db will be paid :)